scipy/signal/_signaltools.py

```diff
@@ -2452,6 +2456,277 @@ def hilbert2(x, N=None):
     return x
 
 
+def envelope(z: np.ndarray, bp_in: tuple[int | None, int | None] = (1, None), *,
+             n_out: int | None = None, squared: bool = False,
+             residual: Literal['lowpass', 'all', None] = 'lowpass',
+             axis: int = -1) -> tuple[np.ndarray, np.ndarray] | np.ndarray:
```

Comment:
It would be quite nice if mutating the return type based on the inputs could be avoided, either by removing the option to have `residual=None` or by returning results object instead

Reply:
I changed the return type to be a single ``np.ndarray``.